config/v1/types_ingress.go

```diff
@@ -32,6 +32,12 @@ type IngressSpec struct {
 	//
 	// Once set, changing domain is not currently supported.
 	Domain string `json:"domain"`
+
+	// tlsSecurityProfile specifies settings for TLS connections for ingresscontrollers.
+	//
+	// If unset, a default (which may change between releases) is chosen.
+	// +optional
+	TLSSecurityProfile *TLSSecurityProfile `json:"tlsSecurityProfile,omitempty"`
```

Comment:
I don't think we should do this.  Having a default for ingresscontrollers match the default for the APIServer makes a cluster-wide default work.  Then you only need the value on your operator in this pull.

Reply:
Are you saying apiserver.config.openshift.io is the cluster default ingress should honor in this case?

Reply:
@Miciah what's your take here?

Reply:
I have no objection, just surprised we had this misunderstanding to begin with and makes me wonder what else I'm missing.

Reply:
I find it surprising that `apiserver.config.openshift.io/cluster` has the default value for ingresscontrollers (and whatever future components need configuration for TLS profile).

Reply:
It is not reasonable to have two cluster-wide default values for this. We should have one.  Ingresses may or may not be used (a user could set up a different mechanism), but APIServers will always be present, so we placed it on apiservers.config.openshift.io.  Making a user specify it twice in the common case isn't friendly or useful to them.

Reply:
Sounds fine to me — I had originally interpreted the design as including a cluster default to drive multiple components, but the API being scoped to the apiserver resource is what surprised me. As long as the docs are clear about how the defaults propagate I don't see an issue. @Miciah, WDYT?

Reply:
> Ingresses may or may not be used (a user could set up a different mechanism)

We're talking about `ingress.config/cluster`, right?  That should always be present, and it should have ingress-related cluster configuration.

Reply:
@deads2k do you have more to say here? It sounds like @Miciah advocates for what I have already, and you're advocating for something else, and I need to know how we come to consensus. Is there anybody else you know of with an opinion?

Reply:
@deads2k @knobunc @Miciah I have removed the `ingress.config.openshift.io` change. The operator can use `apiservers.config.openshift.io` instead, and we can discuss that topic separately.